insonmnia/worker/server.go

```diff
@@ -371,18 +372,38 @@ func (m *Worker) cancelDealTasks(deal *pb.Deal) error {
 }
 
 type runningContainerInfo struct {
-	Description Description   `json:"description,omitempty"`
-	Cinfo       ContainerInfo `json:"cinfo,omitempty"`
-	Spec        pb.TaskSpec   `json:"spec,omitempty"`
+	Description descriptionMarshallable   `json:"description,omitempty"`
+	Cinfo       containerInfoMarshallable `json:"cinfo,omitempty"`
+	Spec        pb.TaskSpec               `json:"spec,omitempty"`
+}
+
+type descriptionMarshallable struct {
+	Description
+	Reference reference.Field `json:"Reference"`
+	Networks  []*structs.NetworkSpec
```

Comment:
why can't we just make networks in description public?

Reply:
Description struct has Networks() and Network() methods.